src/Tasks/CodeTaskFactory.cs

```diff
@@ -712,6 +642,73 @@ private void AddReferenceAssemblyToReferenceList(List<string> referenceAssemblyL
                     _log.LogErrorWithCodeFromResources("CodeTaskFactory.CouldNotFindReferenceAssembly", referenceAssembly);
                 }
             }
+
+            static string GetPathFromPartialAssemblyName(string partialName)
+            {
+                string candidateAssemblyLocation = null;
+
+#pragma warning disable 618, 612
+                // Unfortunately Assembly.Load is not an alternative to LoadWithPartialName, since
+                // Assembly.Load requires the full assembly name to be passed to it.
+                // Therefore we must ignore the deprecated warning.
+                Assembly candidateAssembly = Assembly.LoadWithPartialName(partialName);
+                if (candidateAssembly != null)
+                {
+                    candidateAssemblyLocation = candidateAssembly.Location;
+                }
+                else if (NativeMethodsShared.IsMono)
+                {
```

Comment:
Mono failure from something here?

Reply:
Good guess but no! Instead it was 84643d8309--we just didn't have any test assets in that form before.